shipyard/models/turrets.py

```diff
@@ -15,17 +15,20 @@ class Turret:
     def __init__(self, model_type):
         data = get_file_data("hull_turrets.json")
 
-        self.data = data
-        self.name = model_type                                # name of the model
-        self.model = data.get("models").get(model_type)       # model of the turret
-        self.tonnage    = self.model.get("tonnage")           # size of the turret and addons
-        self.addons     = list()                              # list of the addons for this turret
-        self.max_wep    = self.model.get("num_weapons")       # max number of weapons per turret type
-        self.weapons    = list()                              # list of the weapons on this turret
-        self.cost       = self.model.get("cost")              # cost of the turret and addons
+        self.data            = data
+        self.name            = model_type                          # name of the model
+        self.model           = data.get("models").get(model_type)  # model of the turret
+        self.tonnage         = self.model.get("tonnage")           # size of the turret and addons
+        self.addons          = list()                              # list of the addons for this turret
+        self.max_wep         = self.model.get("num_weapons")       # max number of weapons per turret type
+        self.weapons         = list()                              # list of the weapons on this turret
+        self.cost            = self.model.get("cost")              # cost of the turret and addons
 
     def add_addon(self, part):
-        # Grabbing addon info from data
+        """
+        Handles adding an add-on to the turret
+        :param part: Name of the add-on to add
+        """
         addon = self.data.get("addons").get(part)
```

Comment:
Shouldn't we use .append() here, as addons is a list?

Reply:
This is grabbing the add-on data from the json file before appending it later down the function in line 35.

Reply:
Oh it's not self.addon :P I'm dumb